scalalib/src/mill/scalalib/bsp/ScalaMetalsSupport.scala

```diff
@@ -3,23 +3,37 @@ package mill.scalalib.bsp
 import mill.api.experimental
 import mill.{Agg, T}
 import mill.define.Target
+import mill.scalalib.api.Util.isScala3
 import mill.scalalib.{Dep, DepSyntax, ScalaModule}
 
 /*+ Enable some common settings required to properly support Metals Language Server (via BSP). */
 @experimental
 trait ScalaMetalsSupport extends ScalaModule {
 
   /** The semanticDB version to use. It needs to support your configured Scala versions. */
-  def semanticDbVersion: T[String]
+  def semanticDbVersion: T[String] = T { "4.4.30" }
+
   override def scalacPluginIvyDeps: Target[Agg[Dep]] = T {
-    super.scalacPluginIvyDeps() ++ Agg(
-      ivy"org.scalameta:::semanticdb-scalac:${semanticDbVersion()}"
-    )
+    super.scalacPluginIvyDeps() ++ {
+      if (isScala3(scalaVersion())) {
+        Seq.empty
+      } else {
+        Agg(
+          ivy"org.scalameta:::semanticdb-scalac:${semanticDbVersion()}"
+        )
+      }
+    }
   }
 
   /** Adds some options and configures the semanticDB plugin. */
   override def mandatoryScalacOptions: Target[Seq[String]] = T {
-    super.mandatoryScalacOptions() ++ Seq("-Yrangepos", s"-P:semanticdb:sourceroot:${T.workspace}")
+    super.mandatoryScalacOptions() ++ {
+      if (isScala3(scalaVersion())) {
+        Seq("-Xsemanticdb")
```

Comment:
There is also a `-semanticdb-target` that can be set for Scala 3 if we want more fine grained control over where the semanticdb is placed, but the default seems to work here.

Reply:
Can't say much, as I'm not very familiar with semanticdb details, but as we configure it explicitly for Scala 2, we might also do it for Scala 3 to have more consistency. Without any recommendation from semanticdb-side, it's a 50-50 chance we choose the right strategy. :-p

Reply:
Seeing that this is "ScalaMetalsSupport" and it works with Metals I think it should probably be good with the default. If we find that it does't work I'll come back and address it.